torch/quantization/fx/_equalize.py

```diff
@@ -0,0 +1,173 @@
+              and zero_points are set to 1.0 and 0.
+    """
+
+    def __init__(self, input_dtype=torch.quint8, input_qscheme=torch.per_tensor_affine,
+                 input_quant_min=None, input_quant_max=None, weight_dtype=torch.quint8,
+                 weight_qscheme=torch.per_tensor_affine, weight_quant_min=None,
+                 weight_quant_max=None, factory_kwargs=(None, None)) -> None:
+        super(_InputWeightObserver, self).__init__()
+
+        if input_qscheme not in {torch.per_tensor_affine, torch.per_tensor_symmetric}:
+            raise TypeError("Input qscheme must be per-tensor")
+
+        self.input_obs = PerChannelMinMaxObserver(ch_axis=1, dtype=input_dtype,
+                                                  qscheme=input_qscheme,
+                                                  quant_min=input_quant_min,
+                                                  quant_max=input_quant_max,
+                                                  factory_kwargs=factory_kwargs[0])
+
+        self.weight_col_obs = PerChannelMinMaxObserver(ch_axis=1, dtype=weight_dtype,
+                                                       qscheme=weight_qscheme,
+                                                       quant_min=weight_quant_min,
+                                                       quant_max=weight_quant_max,
+                                                       factory_kwargs=factory_kwargs[1])
+
+        self.weight_row_obs = PerChannelMinMaxObserver(ch_axis=0, dtype=weight_dtype,
+                                                       qscheme=weight_qscheme,
+                                                       quant_min=weight_quant_min,
+                                                       quant_max=weight_quant_max,
+                                                       factory_kwargs=factory_kwargs[1])
+
+        self.equalization_scale = torch.empty(0)
+
+    def forward(self, x_orig, w_orig):
+        # TODO: Allow for convoluational layers
+        if not (x_orig.ndim == 2 and w_orig.ndim == 2 and x_orig.shape[1] == w_orig.shape[1]):
+            raise ValueError(
+                "Input and Weight must have the same column dimension. " +
+                f"Found {x_orig.shape} and {w_orig.shape} instead."
+            )
+
+        return self._forward(x_orig, w_orig)
+
+    def _forward(self, x_orig, w_orig):
+        r"""
+        Calculates the min/max values of each input column, weight column, and weight row.
+        """
+
+        x_orig = self.input_obs(x_orig)
+        w_orig = self.weight_col_obs(w_orig)
+        w_orig = self.weight_row_obs(w_orig)
+
+        # Calculate the column indices of the min/max weight in each row
+        num_row, _ = w_orig.shape
+        min_weights_ind = []
+        max_weights_ind = []
+        for i in range(num_row):
+            min_weights_ind.append(torch.where(w_orig[i] == self.weight_row_obs.min_vals[i])[0][0])
```

Comment:
torch.where(cond) is equivalent to torch.nonzero(condition)
https://pytorch.org/docs/stable/generated/torch.nonzero.html#torch.nonzero